src/snapshot/patch_compute.rs

```diff
@@ -224,6 +227,128 @@ fn compute_children_patches(
     }
 }
 
+/// Trait where NaN values must not be treated as different.
+trait Different {
```

Comment:
This whole trait is... not ideal. I would strongly prefer we move the equality check for `Variant` to its own file, and make it a simple function instead of a trait.

I have a function already implemented in my syncback fork [here](https://github.com/UpliftGames/rojo/blob/syncback-internal-release/src/variant_eq.rs) if you want to steal it.

It does require us to pull in `float_cmp` as a crate, but I think that's fine.